extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -0,0 +1,344 @@
+    this.element.appendChild(this.container_);
+
+    this.registerAction('open', this.activate.bind(this));
+  }
+
+  /** @override */
+  layoutCallback() {
+    // DO NOT ADD CODE HERE
+    // layoutCallback for lightbox-viewer is meaningless, lightbox-viewer
+    // doesn't have children, it just manages elements elsewhere in the page in
+    // `open_` `close_` and `updateViewer_` methods.
+  }
+
+  /**
+   * Builds the page mask and appends it to the container.
+   * @private
+   */
+  buildMask_() {
+    dev().assert(this.container_);
+    const mask = this.win.document.createElement('div');
+    mask.classList.add('-amp-lightbox-viewer-mask');
+    this.container_.appendChild(mask);
+  }
+
+  /**
+   * Builds the controls (i.e. Next, Previous and Close buttons) and appends
+   * them to the container.
+   * @private
+   */
+  buildControls_() {
+    const next = this.next_.bind(this);
+    const prev = this.previous_.bind(this);
+    const close = this.close_.bind(this);
+
+    // TODO(aghassemi): i18n and customization. See https://git.io/v6JWu
+    this.buildButton_('Next', 'amp-lightbox-viewer-button-next', next);
+    this.buildButton_('Previous', 'amp-lightbox-viewer-button-previous', prev);
+    this.buildButton_('Close', 'amp-lightbox-viewer-button-close', close);
+  }
+
+  /**
+   * Builds a button and appends it to the container.
+   * @param {!string} label Text of the button for a11y
+   * @param {!string} className Css classname
+   * @param {!function()} action function to call when tapped
+   * @private
+   */
+  buildButton_(label, className, action) {
+    const button = this.win.document.createElement('div');
+
+    button.setAttribute('role', 'button');
+    button.setAttribute('aria-label', label);
+    button.classList.add(className);
+    button.addEventListener('click', action);
+
+    this.container_.appendChild(button);
+  }
+
+  /**
+   * Opens the lightbox-viewer with either the invocation source or
+   * the element referenced by the `id` argument.
+   * Examples:
+   *  // Opens the element tapped.
+   *  on="tap:myLightboxViewer'
+   *
+   *  // Opens the element referenced by elementId
+   *  on="tap:myLightboxViewer.open(id='<elementId>')
+   * @override
+   */
+  activate(invocation) {
+    let target = invocation.source;
+    if (invocation.args && invocation.args.id) {
+      const targetId = invocation.args.id;
+      target = this.win.document.getElementById(targetId);
+      user().assert(target,
+        'amp-lightbox-viewer.open: element with id: %s not found', targetId);
+    }
+    this.open_(target);
+  }
+
+  /**
+   * Opens the lightbox-viewer and displays the given element inside.
+   * @param {!Element} elem Element to lightbox.
+   * @private
+   */
+  open_(elem) {
+    if (this.activeElem_ == elem) {
+      return;
+    }
+
+    this.updateViewer_(elem);
+
+    this.element.style.display = 'block';
+    this.active_ = true;
+
+    this.win.document.documentElement.addEventListener(
```

Comment:
It looks like this can be double bound if we click from one open lightbox to open another lightbox.